pkg/kubelet/dockershim/libdocker/helpers.go

```diff
@@ -60,6 +60,22 @@ func matchImageTagOrSHA(inspected dockertypes.ImageInspect, image string) bool {
 			// hostname or not, we only check for the suffix match.
 			if strings.HasSuffix(image, tag) || strings.HasSuffix(tag, image) {
 				return true
+			} else {
+				t, err := dockerref.ParseNormalizedNamed(tag)
```

Comment:
Add a comment explaining what this code is doing and a TODO to remove the workaround after docker is fixed properly with a link to the issue.

Reply:
actually, a follow up issue to do better-normalized comparisons in all cases would be good.. the existing prefix checks don't look generally correct to me

Reply:
@liggitt yes, hoping @ncdc would be able to help with that. This is his PR effectively :)